frontend/js/src/listens/AddToPlaylist.tsx

```diff
@@ -0,0 +1,200 @@
+                    />,
+                    { toastId: "load-playlists-error" }
+                );
+            }
+        }
+        try {
+            // First, try to load playlists from sessionStorage
+            const storedPlaylists = window.sessionStorage.getItem("listenbrainz_playlists");
+            if(storedPlaylists){
+                setPlaylists(JSON.parse(storedPlaylists));
+                return;
+            }
+        } catch (error) {
+            // There was an issue parsing the stored playlists
+            // Ignore the issue and load playlists as if there were none stored
+            console.error(error);
+        }
+
+        if(currentUser?.auth_token){
+            loadPlaylists();
+        }
+    }, [APIService,currentUser]);
+
+    const addToPlaylist = React.useCallback(async (event:React.MouseEvent<HTMLButtonElement>)=>{
+        try {
+            if(!currentUser?.auth_token){
+                throw new Error("You are not logged in");
+            }
+            const { currentTarget } = event;
+            const { name:playlistName } = currentTarget;
+            const playlistIdentifier = currentTarget.getAttribute("data-playlist-identifier");
+            let trackToSend;
+            if(has(listen,"title")){
+                trackToSend = listen as JSPFTrack;
+            } else {
+                trackToSend = listenToJSPFTrack(listen as Listen);
+            }
+            if( !playlistIdentifier){
+                throw new Error(`No identifier for playlist ${playlistName}`);
+            }
+            const playlistId = playlistIdentifier?.replace(PLAYLIST_URI_PREFIX, "");
+            const status =  await APIService.addPlaylistItems(currentUser.auth_token,playlistId,[trackToSend]);
+            if (status === 200){
+                toast.success(<ToastMsg title="Added track" message={
+                    <>
+                        Successfully added <i>{trackToSend.title}</i> to playlist <a
+                        href={playlistIdentifier}>
+                            {playlistName}
+                        </a>
+                    </>
+                } />, {
+                    toastId: "success-add-track-to-playlist",
+                });
+            } else {
+                throw new Error("Could not add track to playlist");
+            }
+        } catch (error) {
+            toast.error(<ToastMsg title="Error adding track to playlist" message={
+                <>
+                    Could not add track to playlist:<br/>
+                    {error.toString()}
+                </>
+            } />, {
+                toastId: "error-add-track-to-playlist",
+              });
+        }
+    },[listen,currentUser.auth_token]);
+
+    const createPlaylist = React.useCallback(async()=>{
+        let trackToSend;
+        if(has(listen,"title")){
+            trackToSend = listen as JSPFTrack;
+        } else {
+            trackToSend = listenToJSPFTrack(listen as Listen);
+        }
```

Comment:
I don't fully understand how the two cases come to be, but anyway the same code block is needed also above in line 84. Would be good to move this into a helper function.